File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = [0,0]
    
    while s != "1" :
        remove_zero_s = ''  
        answer[0] += 1
        for elem in s:
            if elem == "1":
                remove_zero_s += '1'
            else:
                answer[1] += 1
        c = len(remove_zero_s)
        s = bin(c)[2:]
        
    return answer

File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
        
        while (s.length() > 1){
            answer[0] ++;
            int numOfOne = 0;
            for(int i = 0; i < s.length(); i++){
                char cur = s.charAt(i);
                if (cur == '0'){
                    answer[1]++;
                } else {
                    numOfOne++;
                }
            }
            s = Integer.toString(numOfOne,2);
            
        }
            //temp의 2진법이 s가 됨 
        return answer;
    }
}